components/family-studies/ui/src/main/resources/PhenoTips/FamilyContentTopMenu.xml

```diff
@@ -160,6 +160,15 @@
       );
     }
   });
+
+  // Family External ID uniqueness check
+  ([$('body')]).each(function(element) {
+      element.select('.external_id input').each(function(input) {
+        if (!input.__uniqueness_validator) {
+          input.__uniqueness_validator = new XWiki.widgets.UniquenessValidator(input, "family");
+        }
+      });
+  });
```

Comment:
Semantically, the menu is not the right place for this piece of code. Since it is strictly related to the Family external id, I would put it in a new jsx in the `FamilyRecordField_externalId.xml` document.

Even better, since the backend code only seems to be able to deal with external ids anyway, i would leave this in the UniquenessValidator widget where it was originally, and figure out the entity type (patient or family) from a css class or a hidden input that is generated by the external id uix.